### ISSUE TITLE
Cannot read graph6

### ISSUE THREAD

**Issue by bslavchev** [networkx/networkx#4032]
I have a basic script that I used a year ago to convert graph6 strings into PACE-format graph files; it doesn't work now with networkx version 2.4: 
`in from_graph6_bytes
    if string.startswith(b'>>graph6<<'):
TypeError: startswith first arg must be str or a tuple of str, not bytes`

**Comment by rossbar** [comment]
Perhaps you've switched from Python 2 to Python 3 over the last year? It seems that `from_graph6_bytes` expects a bytes-like object, not a string. An update to the `Parameters` section of the docstring could perhaps make this more clear.

If this is indeed a Python 2/3 issue you could try encoding your Python3 string to produce a sequence of bytes, e.g. something like `from_graph6_bytes(<your_string>.encode('ascii'))` in your script.

**Comment by bslavchev** [comment]
> 
> 
> Perhaps you've switched from Python 2 to Python 3 over the last year? It seems that `from_graph6_bytes` expects a bytes-like object, not a string. An update to the `Parameters` section of the docstring could perhaps make this more clear.
> 
> If this is indeed a Python 2/3 issue you could try encoding your Python3 string to produce a sequence of bytes, e.g. something like `from_graph6_bytes(<your_string>.encode('ascii'))` in your script.

I believe that the error is coming from within NetworkX's code - the call to `string.startswith()` with is made within the `from_graph6_bytes` function. 

Or maybe... I'm passing a String object to it, which it is internally naming 'string' but expecting that it is actually a sequence of bytes? And String's `startswith` function expects a String argument, whereas Byte's expects a Byte argument? Does that make sense?

**Comment by rossbar** [comment]
> I believe that the error is coming from within NetworkX's code - the call to string.startswith() with is made within the from_graph6_bytes function.

This is what I was referring to before - it's confusing because the variable `string` in the function `from_graph6_bytes` is not actually a string - there is a mismatch between the function name (which implies that the input should be bytes) and the naming convention *inside* the function (argument name is string). This is likely an artifact from when the function was originally written (pre-Python3) as the naming convention would have made sense for Python 2.

Here's a quick example to illustrate the root of the problem:

```python
>>> s = '>>graph6<<A_'
>>> type(s)
str
>>> b = s.encode('ascii')
>>> type(b)
bytes
>>> type(b'>>graph6<<')
bytes
# This produces the error because the str.startswith method expects strings,
# not bytes
>>> s.startswith(b'>>graph6<<')
Traceback (most recent call last)
   ...
TypeError: startswith first arg must be str or a tuple of str, not bytes
# But a comparison of bytes to bytes is fine
>>> b.startswith(b'>>graph6<<')
True
```
And with `nx.from_graph6_bytes`:
```python
>>> import networkx as nx
>>> nx.from_graph6_bytes(s)
Traceback (most recent call last)
   ...
TypeError: startswith first arg must be str or a tuple of str, not bytes
>>> G =  nx.from_graph6_bytes(b)
>>> G.edges
EdgeView([(0, 1)])
```

Long story short - `from_graph6_bytes` needs a bytes-like input, *not* a string.

I think updating the argument name, docstring, and maybe touching up the exceptions would help resolve some of this confusion.

**Comment by bslavchev** [comment]
Right, I believe you are correct! I converted the string into bytes, which solves the issue. Thanks for the help, and for fixing the issue :)
